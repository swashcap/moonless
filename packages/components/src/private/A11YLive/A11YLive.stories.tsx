import React from 'react';

import { A11YLive } from './A11YLive';

export default {
  component: A11YLive,
  title: 'Private/A11YLive',
};

export const Default = () => <A11YLive />;

export const WithAssertive = () => <A11YLive assertive="This is assertive." />;

export const WithPolite = () => <A11YLive polite="This is polite." />;
