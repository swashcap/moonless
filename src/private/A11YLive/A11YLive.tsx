import React from 'react';

import { VisuallyHidden } from '../../VisuallyHidden/VisuallyHidden';

export interface A11YLiveProps {
  assertive?: React.ReactNode;
  polite?: React.ReactNode;
}

export const A11YLive: React.FC<A11YLiveProps> = ({ assertive, polite }) => (
  <VisuallyHidden>
    <div aria-live="assertive">{assertive}</div>
    <div aria-live="polite">{polite}</div>
  </VisuallyHidden>
);
