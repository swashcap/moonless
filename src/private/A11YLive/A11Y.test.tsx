import React from 'react';
import { render } from '@testing-library/react';

import { A11YLive } from './A11YLive';

describe('A11YLive', () => {
  test('It should render.', () => {
    const { container } = render(<A11YLive />);

    expect(container).toMatchSnapshot();
  });

  test('It should render live regions.', () => {
    const { container } = render(<A11YLive />);

    expect(container.querySelector('[aria-live="assertive"]')).toBeTruthy();
    expect(container.querySelector('[aria-live="polite"]')).toBeTruthy();
  });

  test('It should pass assertive property.', () => {
    const { container } = render(<A11YLive assertive="This is assertive." />);

    expect(
      container.querySelector('[aria-live="assertive"]')?.textContent
    ).toBe('This is assertive.');
  });

  test('It should pass polite property.', () => {
    const { container } = render(<A11YLive polite="This is polite." />);

    expect(container.querySelector('[aria-live="polite"]')?.textContent).toBe(
      'This is polite.'
    );
  });
});
