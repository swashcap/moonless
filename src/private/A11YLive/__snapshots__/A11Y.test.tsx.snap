// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A11YLive It should render. 1`] = `
<div>
  <span
    class="VisuallyHidden"
  >
    <div
      aria-live="assertive"
    />
    <div
      aria-live="polite"
    />
  </span>
</div>
`;
